src/sql/workbench/browser/modelComponents/declarativeTable.component.ts

```diff
@@ -119,14 +92,22 @@ export default class DeclarativeTableComponent extends ComponentBase implements
 	}
 
 	public onHeaderCheckBoxChanged(e: boolean, colIdx: number): void {
-		this.data.forEach((row, rowidx) => {
-			if (row[rowidx][colIdx] !== e) {
-				this.onCellDataChanged(e, rowidx, colIdx);
+		this.data.forEach((row, rowIdx) => {
+			if (row[colIdx] !== e) {
+				this.onCellDataChanged(e, rowIdx, colIdx);
 			}
 		});
 		this._changeRef.detectChanges();
 	}
 
+	public trackByFnRows(index: number, item: any): any {
```

Comment:
BTW, I am also making changes in this component right now to fix some other problem.
IIUC, this is incorrect, for example, if I update the data with the same number of rows, the table won't be rendered,

Reply:
I haven't tested it extensively, can you give me a more detailed example?

Reply:
this is my pr: https://github.com/microsoft/azuredatastudio/pull/10833, i am fixing something else there but also improved the frequency it refreshes the UI components, I will merge it now and please sync to see whether it helps.

to fully test your change, please update the table's data multiple times and see whether it handles that correctly.